WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/EditableOrderViewModelTests.swift

```diff
@@ -416,6 +416,34 @@ final class EditableOrderViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.selectedProductsAndVariationsIDs.count, 2)
     }
 
+    func test_clearSelectedItems_clears_selectedProducts_and_selectedProductVariations() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isProductMultiSelectionM1Enabled: true)
+        let product = Product.fake().copy(siteID: sampleSiteID, productID: sampleProductID, productTypeKey: "variable", purchasable: true, variations: [20])
+        let productVariation = ProductVariation.fake().copy(siteID: sampleSiteID,
+                                                            productID: sampleProductID,
+                                                            productVariationID: 20,
+                                                            sku: "product-variation", purchasable: true)
+        storageManager.insertSampleProduct(readOnlyProduct: product)
+        storageManager.insertSampleProductVariation(readOnlyProductVariation: productVariation, on: product)
+        let viewModel = EditableOrderViewModel(siteID: sampleSiteID, stores: stores, storageManager: storageManager, featureFlagService: featureFlagService)
+
+        // When
+        viewModel.isProductMultiSelectionBetaFeatureEnabled = true
+        viewModel.productSelectorViewModel.selectProduct(product.productID)
+        viewModel.selectedProductVariations.append(productVariation)
```

Comment:
Nit: should we keep `selectedProducts` and `selectedProductVariations` readonly with `private(set)`, and test this by calling `viewModel.productSelectorViewModel.getVariationsViewModel?().selectVariation` instead? That would protect the arrays from unexpected updates from outside.

Reply:
Much better, yes. Changed here: dbbb4f4c6255caec32351ae553fa22221d67e8c3

Reply:
With this change I didn't realize that `test_selectedProductsAndVariationsIDs_keeps_track_of_products_and_variations_added_to_the_order` would fail because of the internals of `getVariationsViewModel()`, however there's no reason either for keeping `selectedProductsAndVariationsIDs` accessible, as is a mere counter of the previous too. 

Made it private and removed the extra test here: https://github.com/woocommerce/woocommerce-ios/pull/9240/commits/6e413f34e180e03d59ac764c71eadf092a53c072